src/main/java/com/aws/greengrass/logmanager/model/LogFileGroup.java

```diff
@@ -82,4 +98,21 @@ public void forEach(Consumer<LogFile> callback) {
     public boolean isEmpty() {
         return this.logFiles.isEmpty();
     }
+
+    /**
+     * Get the LogFile object from the fileHash.
+     * @param fileHash the fileHash obtained from uploader.
+     * @return the logFile.
+     */
+    public LogFile getFile(String fileHash) {
```

Comment:
Return an Optional<Logfile> that represents something might not be there. If you do that you dont need the log